dolphinscheduler-task-plugin/dolphinscheduler-task-flink/src/main/java/org/apache/dolphinscheduler/plugin/task/flink/FlinkArgsUtils.java

```diff
@@ -164,12 +164,9 @@ public static List<String> buildInitOptionsForSql(FlinkParameters flinkParameter
             }
 
             // yarn.application.queue
-            String others = flinkParameters.getOthers();
-            if (StringUtils.isEmpty(others) || !others.contains(FlinkConstants.FLINK_QUEUE)) {
-                String queue = flinkParameters.getQueue();
-                if (StringUtils.isNotEmpty(queue)) {
-                    initOptions.add(String.format(FlinkConstants.FLINK_FORMAT_YARN_APPLICATION_QUEUE, queue));
-                }
+            String queue = flinkParameters.getQueue();
+            if (StringUtils.isNotEmpty(queue)) {
```

Comment:
We will check whether user defines this arg by themselves in `others`, so you just have to add one condition in the original way: `!others.contains(FlinkConstants. FLINK_QUEUE_FOR_TARGETS)`, btw, may I ask why you name them `FLINK_QUEUE_FOR_MODE` and `FLINK_QUEUE_FOR_TARGETS `?

Reply:
The YARN queue should be assigned by property `yarn.application.queue` rather than -yqu option, this option is not available in sql-client.sh.  If we want to specify the YARN queue used by a specific Flink SQL task, it would be more appropriate to have an explicit queue option in the task submission form rather than relying on parameters in the 'others' section. Regarding the naming here, it is because in flink-run, the -yqu option only takes effect within the -m option (for mode). When using the -t option (for target), it is necessary to specify it using -Dyarn.application.queue=%s.

Reply:
It seems that we don't have any option for setting yarn queue, `flinkParameters.queue` will be assigned by `taskExecutionContext.queue`, which pass from `processInstance`, however, it's never be assigned.

Reply:
The execution of a workflow is tied to a specific tenant, and this tenant holds a queue attribute. This attribute is assigned to the processInstance being executed by the tenant. The queue attribute of the executionContext for any taskInstance belonging to this processInstance will also be consistent. Therefore, the queue of a task is ultimately determined by the queue attribute of the runtime tenant if not explicitly specified.

Reply:
It seems that this logic is missing, I've debugged and find the queue of `processInstance` is null, also in codes, I don't find where it's assigned by tenant's queue. Would like to help check in your local env?

Reply:
Sure. From my practice, the queue is assigned. This logic should be correctly implemented in the current version.

Reply:
It's weird, do you test on branch dev?

Reply:
Have you tested the modification in this PR on branch dev?

Reply:
I will test it tomorrow, in my env I cp this pr to my 3.1.7.